### PR TITLE
Fix cross-region decryption bug in Java and C#

### PR DESCRIPTION
**Pull request by sushantmimani**
- Update logic to check the start of keyId instead of the entire id for partitioned keys
- Add regression tests for cross-region encryption/decryption

### CODE REVIEW

**Review by tarkatronic — APPROVED** [review]

File: csharp/AppEncryption/AppEncryption.IntegrationTests/Regression/DynamoDbGlobalTableTest.cs [hunk 1]
```diff
@@ -77,6 +66,20 @@ public void Dispose()
...
+
+        if (withKeySuffix)
+        {
+          builder = builder.WithKeySuffix();
+        }
+
+        DynamoDbMetastoreImpl dynamoDbMetastore = builder.Build();
+        return SessionFactoryGenerator.CreateDefaultSessionFactory(configFixture.KeyManagementService, dynamoDbMetastore);
+      }
+
```
Comment (aka-bo) [inline csharp/AppEncryption/AppEncryption.IntegrationTests/Regression/DynamoDbGlobalTableTest.cs:82]: whitespace

File: java/app-encryption/src/it/java/com/godaddy/asherah/regression/DynamoDbGlobalTableIT.java [hunk 2]
```diff
@@ -65,52 +66,72 @@ public void teardown() throws Exception {
...
     }
 
-    DynamoDbMetastoreImpl dynamoDbMetastore = builder.build();
-    return SessionFactoryGenerator.createDefaultSessionFactory(TestSetup.createKeyManagemementService(), dynamoDbMetastore);
+    // Decrypt dataRowString with metastore with region suffix
+    sessionFactory = getSessionFactory(true, REGION_SUFFIX);
+    try (Session<byte[], byte[]> sessionBytes = sessionFactory.getSessionBytes("shopper12345")) {
+      decryptedBytes = sessionBytes.decrypt(dataRowRecordBytes);
+    }
+    assertTrue(Arrays.equals(decryptedBytes, originalBytes));
```
Comment (aka-bo) [inline java/app-encryption/src/it/java/com/godaddy/asherah/regression/DynamoDbGlobalTableIT.java:118]: whitespace issues here as well

**Review by aka-bo — CHANGES_REQUESTED** [review]
in need of a couple of whitespace touch-ups but looks good otherwise

**Review by aka-bo — APPROVED** [review]
LGTM

**Pull request merged**